plugins/modules/packaging/os/rpm_ostree_pkg.py

```diff
@@ -0,0 +1,174 @@
+      - Name of overlay package to install / remove.
+      required: true
+      type: list
+      elements: str
+      aliases: [ pkg ]
+    state:
+      description:
+      - state of the overlay package.
+      - C(present) and C(installed) will simply ensure that a desired package is installed.
+      - C(absent) and C(removed) will remove the specified package.
+      choices: [ 'absent', 'installed', 'present', 'removed' ]
+      default: 'present'
+      type: str
+author:
+- Dusty Mabe (@dustymabe)
+- Abhijeet Kasurde (@Akasurde)
+'''
+
+EXAMPLES = r'''
+- name: Install overlay package
+  community.general.rpm_ostree_pkg:
+    name: nfs-utils
+    state: present
+
+- name: Remove overlay package
+  community.general.rpm_ostree_pkg:
+    name: nfs-utils
+    state: absent
+'''
+
+RETURNS = r'''
+rc:
+    description: Return code of rpm-ostree command
+    returned: always
+    type: int
+    sample: 0
+changed:
+    description: State changes
+    returned: always
+    type: bool
+    sample: True
+action:
+    description: Action performed
+    returned: always
+    type: str
+    sample: 'install'
+packages:
+    description: A list of packages specified
+    returned: always
+    type: list
+    sample: ['nfs-utils']
+stdout:
+    description: Stdout of rpm-ostree command
+    returned: always
+    type: str
+    sample: 'Staging deployment...done\n...'
+stderr:
+    description: Stderr of rpm-ostree command
+    returned: always
+    type: str
+    sample: ''
+cmd:
+    description: Full command used for performed action
+    returned: always
+    type: str
+    sample: 'rpm-ostree uninstall --allow-inactive --idempotent --unchanged-exit-77 nfs-utils'
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+class RpmOstreePkg:
+    def __init__(self, module):
+        self.module = module
+        self.params = module.params
+        self.state = module.params['state']
+
+    def ensure(self):
+        results = dict(
+            rc=0,
+            changed=False,
+            action='',
+            packages=[],
+            stdout='',
+            stderr='',
+            cmd='',
+        )
+
+        # Ensure rpm-ostree command exists
+        cmd = [self.module.get_bin_path('rpm-ostree', required=True)]
+
+        # Decide action to perform
+        if self.state in ('installed', 'present'):
+            results['action'] = 'install'
+            cmd.append('install')
+        elif self.state in ('absent', 'removed'):
+            results['action'] = 'uninstall'
+            cmd.append('uninstall')
+
+        # additional parameters
+        cmd.extend(['--allow-inactive', '--idempotent', '--unchanged-exit-77'])
+        for pkg in self.params['name']:
+            cmd.append(pkg)
+            results['packages'].append(pkg)
+
+        rc, out, err = (0, '', '')
+        if not self.module.check_mode:
+            rc, out, err = self.module.run_command(cmd)
+
+        results.update(dict(
+            rc=rc,
```

Comment:
```suggestion
        rc, out, err = (0, '', '')
        if not self.module.check_mode:
            rc, out, err = self.module.run_command(cmd)

        results.update(dict(
            rc=rc,
```
but if it's already installed, the module in check_mode will return changed=True anyway?
can we check anyhow whether a package is already installed?

Reply:
rpm-ostree provides `--idempotent` parameter which checks it already installed packages for us and returns err code 77 for unchanged.

Reply:
ok, in check_mode it won't be checked, will it?

Reply:
check_mode will always return changed=True, yes?

Reply:
I am afraid, yes, since there is no command to query existing package (atleast I didn't find).

Reply:
Maybe in that case the module should not advertise it supports check mode?